### PR TITLE
Capstone project

### PR DESCRIPTION
**Pull request by SekoViper**
# Summary
## Implement the following: 

- When the page loads, the app shows the list of products
- filtered out products with categories
- When the user clicks (or taps) on a category item, the application navigates to the details page.
- Add Testing for the app

Link to the project
[Project recording](https://drive.google.com/file/d/1Vs8n5iWmJ9bMK5fjnecNbjfO2lvCWont/view?usp=sharing)

### CODE REVIEW

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
```markdown
Hi Seko, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not follow Git flow and linter not configured. Kindly follow Git flow for development and also make sure that PR has all the linter checks green.
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

Invalid Code Review Request **does not count** into the code reviews limit._
```

File: src/__test__/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { render } from '@testing-library/react';
+import { useSelector } from 'react-redux';
+import Products from '../Components/Products/Products';
+import ProductsPage from '../Components/Products/ProductsPage';
+
+jest.mock('react-redux');
+jest.mock('react-router-dom');
+
```
Comment (OLIPLICHE) [inline src/__test__/Home.test.js:8]: - Please add unit tests for some pure functions in your Redux code. It is part of the requirements

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @SekoViper, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit has a green check mark (✅)) 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used 👏

## Required Changes ♻️

- [ ] **Video presentation**:
- Please record a video presenting the features of the project.
- The maximum length of the video is 5 minutes.
- Make sure the video contains a highlight about a piece of the code you find interesting or proud of making.

- [ ] **Design**: Please you need to follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including:
- Typography: font face, size, and weight
- Layout: composition and space between elements.

- See your Design: 👇 
- 
Capture1
- See the Original design: 👇 

-
_**Check the comments under the review**._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Home.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { render } from '@testing-library/react';
+import { useSelector } from 'react-redux';
+import Products from '../Components/Products/Products';
+import ProductsPage from '../Components/Products/ProductsPage';
+import store from '../Redux/configureStore';
+
+jest.mock('react-redux');
+jest.mock('react-router-dom');
+
```
Comment (tresorsawasawa) [inline src/__test__/Home.test.js:9]: - [ ] As mentioned by the previous reviewer, kindly make sure you add **unit tests for some pure functions** in your Redux code. It is part of the requirements

File: src/Components/Products/Products.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                {' '}
+                {product.price}
+              </p>
+            </div>
+          </NavLink>
+        </div>
+      )) }
+    </div>
+  </div>
+);
```
Comment (tresorsawasawa) [inline src/Components/Products/Products.jsx:29]: - [ ] As mentioned by the previous reviewer, kindly make sure you follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including:
  -  Typography: font face, size, and weight
  -  Layout: composition and space between elements.
**Here is how look the home pages :point_down:**

| Template design | Your Design |
| --------------------- | ------------------- |
| Screenshot from 2023-01-12 23-41-09 | Screenshot from 2023-01-12 23-40-09 |

File: src/Components/Products/ProductDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            {' '}
+            {product.price}
+            .00
+          </p>
+          <p>{product.description}</p>
+          <button type="button" className="rounded-none bg-green-700 px-2 py-1 my-3 uppercase font-bold text-white">Add To Cart</button>
+        </div>
+      </div>
+
+    </div>
```
Comment (tresorsawasawa) [inline src/Components/Products/ProductDetails.jsx:36]: - [ ] As mentioned by the previous reviewer, kindly make sure you follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including:
  -  Typography: font face, size, and weight
  -  Layout: composition and space between elements.
**Here is how look the detail pages :point_down:** 

| Template design | Your Design |
| --------------------- | ------------------- |
| Screenshot from 2023-01-12 23-41-24 | Screenshot from 2023-01-12 23-45-50 |

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @SekoViper 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions
-  [**OPTIONAL**] **Kindly make sure for your next solo projects where you are asked Gitflow to open you PR from a `feature` branch to the `dev` and not from the `dev` branch to `main`**. 
You can read this **[article](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)** for a short remind about **[Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)**

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 5]
```diff
@@ -4,18 +4,50 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
-    "dev": "vite",
+    "dev": "vite --open --host",
+    "preview": "vite preview",
     "build": "vite build",
-    "preview": "vite preview"
+    "test": "react-scripts test"
```
Comment (DammyShittu) [inline package.json:10]: - [ ] Kindly add more tests as suggested earlier.

- [ ] I am unable to run your existing tests. When I run `npm test` I get this error message. Kindly fix that.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @SekoViper,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 5 above)
Comment (frank1738) [inline package.json:10]: - [ ] The error still persists when I try to test your application. Kindly fix that

image

Comment (SekoViper) [reply package.json:10]: Hello @frank1738 ,
Kindly re-install react scripts, I hope that solves the problem at your end.  (npm install react-scripts --save)
The test cases run well at my end even as I clone the project again.
image

Comment (frank1738) [reply package.json:10]: @SekoViper  Okey. Solved. Kindly resubmit your project for approval

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @SekoViper           :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**